From 1f204099f3657541772201dda1031fafcb92f502 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Wed, 29 Aug 2012 14:08:00 +0800
Subject: [PATCH 345/449] ENGR00221848 doc: usb: device: Add pc send
 suspend/resume test doc

Add PC send suspend/resume to device board test

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 test/doc/ut_um.toc                       |  1 +
 test/mxc_usb_test/doc/FSL-UT-USB-018.txt | 48 ++++++++++++++++++++++++++++++++
 2 files changed, 49 insertions(+)
 create mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-018.txt

diff --git a/test/doc/ut_um.toc b/test/doc/ut_um.toc
index a3beead..ad6c52e 100644
--- a/test/doc/ut_um.toc
+++ b/test/doc/ut_um.toc
@@ -150,6 +150,7 @@ include::../mxc_usb_test/doc/FSL-UT-USB-014.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-015.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-016.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-017.txt[]
+include::../mxc_usb_test/doc/FSL-UT-USB-018.txt[]
 
 V4L2
 ----
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-018.txt b/test/mxc_usb_test/doc/FSL-UT-USB-018.txt
new file mode 100644
index 0000000..6317949
--- /dev/null
+++ b/test/mxc_usb_test/doc/FSL-UT-USB-018.txt
@@ -0,0 +1,48 @@
+FSL-UT-USB-018-PC-Send-Suspend-Resume-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Using pc to send suspend and resume signal device
+
+| Automated |
+Yes
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+1. load one usb gadget driver, for example
+modprobe g_file_storage file=/dev/mmcblk0p1 removable=1
+2. Connect the board with Linux PC using usb cale
+NOTE: Linux PC is must, as Windows PC will send reset after
+resume.
+3. enable USB gadget wakeup.
+For otg port
+
+    echo enabled > /sys/devices/platform/fsl-usb2-otg/power/wakeup
+
+For device-only port
+
+    echo enabled > /sys/devices/platform/fsl-usb2-udc/power/wakeup
+
+4. Suspend Linux PC
+5. After Linux PC suspend, using "echo standby > /sys/power/state"
+to let the board to suspend
+6. Resume Linux PC
+7. There is no re-enumeration occurs, and the storage can be mounted.
+
+| Expected Result |
+No disconnect and re-enumeration occurs after pc sends resume.
+
+|====================================================================
+
+<<<
-- 
1.8.0

